### ISSUE TITLE
rebar3 geas is hanging

### ISSUE THREAD

**Issue by walter-weinmann** [crownedgrouse/geas_rebar3#6]
The following project https://github.com/walter-weinmann/ocparse has problems with rebar3 geas.

```
D:\SoftDevelopment\Projects\ocparse_idea\ocparse>rebar3 geas
===> Expanded command sequence to be run: [{default,app_discovery},
                                           {default,install_deps},
                                           {default,lock},
                                           {default,compile},
                                           {default,geas}]
===> Not adding provider default geas from module geas_rebar3 because it already exists from module geas_rebar3
===> Not adding provider default geas from module geas_rebar3 because it already exists from module geas_rebar3
===> Verifying dependencies...
===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", pre_hooks, compile) -> no hooks defined

===> sh info:
        cwd: "d:/SoftDevelopment/Projects/ocparse_idea/ocparse"
        cmd: cmd /c mklink /j "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\_build\\default\\lib\\ocparse\\priv" "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\priv"

===>    opts: [{use_stdout,false},return_on_error]

===> Port Cmd: cmd /q /c cmd /c mklink /j "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\_build\\default\\lib\\ocparse\\priv" "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\priv"
Port Opts: [exit_status,{line,16384},use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/SoftDevelopment/Projects/ocparse_idea/ocparse"
        cmd: cmd /c mklink /j "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\_build\\default\\lib\\ocparse\\src" "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\src"

===>    opts: [{use_stdout,false},return_on_error]

===> Port Cmd: cmd /q /c cmd /c mklink /j "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\_build\\default\\lib\\ocparse\\src" "d:\\SoftDevelopment\\Projects\\ocparse_idea\\ocparse\\src"
Port Opts: [exit_status,{line,16384},use_stdio,stderr_to_stdout,hide,eof]

===> Compiling ocparse
===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", pre_hooks, compile) -> no hooks defined

===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", pre_hooks, erlc_compile) -> no hooks defined

===>      Skipped oclexer.xrl
d:/SoftDevelopment/Projects/ocparse_idea/ocparse/_build/default/lib/ocparse/src/ocparse.yrl: Warning: conflicts: 19 shift/reduce, 0 reduce/reduce
d:/SoftDevelopment/Projects/ocparse_idea/ocparse/_build/default/lib/ocparse/src/ocparse.yrl:none: Warning: conflicts: 19 shift/reduce, 0 reduce/reduce

===>      Compiled ocparse.yrl
===> erlopts [debug_info]
===> files to compile ["d:/SoftDevelopment/Projects/ocparse_idea/ocparse/_build/default/lib/ocparse/src/ocparse_util.erl",
                       "d:/SoftDevelopment/Projects/ocparse_idea/ocparse/_build/default/lib/ocparse/src/ocparse_generator.erl",
                       "d:/SoftDevelopment/Projects/ocparse_idea/ocparse/_build/default/lib/ocparse/src/ocparse.erl",
                       "d:/SoftDevelopment/Projects/ocparse_idea/ocparse/_build/default/lib/ocparse/src/oclexer.erl"]
===>      Compiled ocparse.erl
===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", post_hooks, erlc_compile) -> no hooks defined

===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", pre_hooks, app_compile) -> no hooks defined

===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", post_hooks, app_compile) -> no hooks defined

===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", post_hooks, compile) -> no hooks defined

===> run_hooks("d:/SoftDevelopment/Projects/ocparse_idea/ocparse", post_hooks, compile) -> no hooks defined
```

**Comment by crownedgrouse** [comment]
I will check, thanks for reporting.

**Comment by crownedgrouse** [comment]
Sorry cannot reproduce, on Linux with latest rebar3 : 

`eric@crownedgrouse ~/tmp/ocparse $ ./rebar3 geas
===> Fetching geas_rebar3 ({git,
                                      "https://github.com/crownedgrouse/geas_rebar3.git",
                                      {branch,"master"}})
===> Fetching geas ({git,"git://github.com/crownedgrouse/geas.git",
                                {branch,"master"}})
===> Compiling geas
===> Compiling geas_rebar3
===> Fetching geas_rebar3 ({git,
                                      "https://github.com/crownedgrouse/geas_rebar3.git",
                                      {branch,"master"}})
===> Fetching geas ({git,"git://github.com/crownedgrouse/geas.git",
                                {branch,"master"}})
===> Compiling geas
===> Compiling geas_rebar3
===> Verifying dependencies...
===> Compiling ocparse
/home/eric/tmp/ocparse/_build/default/lib/ocparse/src/ocparse.yrl: Warning: conflicts: 25 shift/reduce, 0 reduce/reduce
/home/eric/tmp/ocparse/_build/default/lib/ocparse/src/ocparse.yrl:none: Warning: conflicts: 25 shift/reduce, 0 reduce/reduce

   R15                   20.1       Geas database       
---Min--------Arch-------Max----------------------------------------------------
   18.0                             ocparse              
                                    geas_rebar3          
   18.0                             ocparse              
--------------------------------------------------------------------------------
   18.0                  20.1       Global project       

./_build/default/lib/ocparse/ebin/ocparse_generator.beam
18.0      erlang:monotonic_time/1
`
looks like you are running a Window machine, please precise what version in order I try to test in same condition.

**Comment by walter-weinmann** [comment]
I'm running Windows 10.

```
C:\Users\walte>rebar3 -v
rebar 3.4.4 on Erlang/OTP 20 Erts 9.1

```

**Comment by crownedgrouse** [comment]
`rebar 3.4.3 on Erlang/OTP 20 Erts 9.1` on my side. I do not have a Windows 10 right now. Need more time.

**Comment by crownedgrouse** [comment]
`rebar 3.4.3 on Erlang/OTP 20 Erts 9.1` on my side. I do not have a Windows 10 right now. Need more time.

**Comment by crownedgrouse** [comment]
I could find a Windows 10 machine.
I installed from scratch Erlang, git, rebar3  and could apply geas on ocparse project.
```
C:\Users\Public\Eric\ocparse>rebar3 geas
===> Fetching geas_rebar3 ({git,"https://github.com/crownedgrouse/geas_rebar3.git",
                                {branch,"master"}})
===> Fetching geas ({git,"git://github.com/crownedgrouse/geas.git",
                         {branch,"master"}})
===> Compiling geas
===> Compiling geas_rebar3
===> Verifying dependencies...
===> Compiling ocparse
   R15                   20.1       Geas database
---Min--------Arch-------Max----------------------------------------------------
   18.0                             ocparse
                                    geas_rebar3
   18.0                             ocparse
--------------------------------------------------------------------------------
   18.0                  20.1       Global project

./_build/default/lib/ocparse/ebin/ocparse_generator.beam
18.0      erlang:monotonic_time/1

./ebin/ocparse_generator.beam
18.0      erlang:monotonic_time/1
```
Please considere testing from my former comment, and also by moving plugins declaration to `%userprofile%\.config\rebar3\rebar.config` instead local rebar.config.
I wait your feedback before thinking to close the ticket.

**Comment by crownedgrouse** [comment]
Hi, I see a plugin issue in rebar 3.4.4 https://github.com/erlang/rebar3/issues/1657  so maybe you should try 3.4.3 that worked for me.

**Comment by walter-weinmann** [comment]
Version 3.4.3 didn't solve my problem - same with 3.4.6 but it is running fine in Travis.

**Comment by crownedgrouse** [comment]
Is D:\   a local mount point or a network share at your end ? If running fine on Travis, this is something wrong locally ?

**Comment by walter-weinmann** [comment]
D:\ is a local Windows drive.

Travis is running with some Unix background - I suppose my problem is somehow caused by the Windows environment. I have a very similar project (syparse) with no problem at all.

It is all very strange.

**Comment by crownedgrouse** [comment]
try to dist-clean and fetch again deps. rebar is using some cache. If your other project work fine in D:\  there is no reason other fails then. 
unfortunately I cannot help more as I cannot reproduce. I'm afraid you will need to find issue your side. I do not close the issue for now until I'm not sure geas is not guilty.

**Comment by crownedgrouse** [comment]
Just jumping to my eyes : do you other project {use_stdout,false} in rebar opts ?  geas need to write on stdout for result...

**Comment by crownedgrouse** [comment]
I do not see this option in the rebar.config file coming from your project git clone

**Comment by crownedgrouse** [comment]
https://www.rebar3.org/v3/discuss/59afc0c2e30286001bb8cc8a  see also rebar3 report, and this issue looks like yours

**Comment by walter-weinmann** [comment]
Fixed with rebar 3.4.7.

Thank you @crownedgrouse  for the helpful support - this is really a great tool.

**Comment by crownedgrouse** [comment]
Very kind support of this tool ! Thanks !
